crates/torii/indexer/src/processors/erc1155_transfer_single.rs

```diff
@@ -0,0 +1,79 @@
+use anyhow::Error;
+use async_trait::async_trait;
+use cainome::cairo_serde::{CairoSerde, U256 as U256Cainome};
+use dojo_world::contracts::world::WorldContractReader;
+use starknet::core::types::{Event, U256};
+use starknet::providers::Provider;
+use torii_sqlite::Sql;
+use tracing::debug;
+
+use super::{EventProcessor, EventProcessorConfig};
+use crate::task_manager::{self, TaskId, TaskPriority};
+
+pub(crate) const LOG_TARGET: &str = "torii_indexer::processors::erc1155_transfer_single";
+
+#[derive(Default, Debug)]
+pub struct Erc1155TransferSingleProcessor;
+
+#[async_trait]
+impl<P> EventProcessor<P> for Erc1155TransferSingleProcessor
+where
+    P: Provider + Send + Sync + std::fmt::Debug,
+{
+    fn event_key(&self) -> String {
+        "TransferSingle".to_string()
+    }
+
+    fn validate(&self, event: &Event) -> bool {
+        // key: [hash(TransferSingle), operator, from, to]
+        // data: [id.low, id.high, value.low, value.high]
+        if event.keys.len() == 4 && event.data.len() == 4 {
+            return true;
+        }
+        false
```

Comment:
```suggestion
        event.keys.len() == 4 && event.data.len() == 4
```
Since we have it twice, may be a function reused here?